tests/test_utils.py

```diff
@@ -213,3 +213,27 @@ def test_merge_meta():
 
     obs = utils.merge_meta(left, right, ignore_meta_conflict=True)
     pdt.assert_frame_equal(exp, obs)
+
+
+def test_get_variable_components_int():
+    assert utils.get_variable_components('foo|bar|baz', 1) == ['bar']
```

Comment:
I was a bit surprised why calling the function with an integer would return a list, not the string. Then I realized that this line 
```python
        level = [level] if type(level) == int else level
```
in the function turn the `level` arg into a list and makes the second part of 
```python
        return [_x[i] for i in level] if islistable(level) else _x[level]
```
obsolete (because `level` will always be a list at this point).

If you agree that the returning the string makes more sense, moving the cast-to-list line into the `else` section would do the trick...

Reply:
I agree. And by the way a very nice way to internalize the git workflow :)